elasticapm/conf/__init__.py

```diff
@@ -10,8 +10,121 @@
+
+
+class _ListConfigValue(_ConfigValue):
+    def __init__(self, dict_key, list_separator=',', **kwargs):
+        self.list_serparator = list_separator
+        super(_ListConfigValue, self).__init__(dict_key, **kwargs)
+
+    def __set__(self, instance, value):
+        if isinstance(value, compat.string_types):
+            value = value.split(self.list_serparator)
+        else:
+            value = list(value) if value is not None else value
+        if value:
+            value = [self.type(item) for item in value]
+        instance._values[self] = value
+
+
+class _BoolConfigValue(_ConfigValue):
+    def __init__(self, dict_key, true_string='true', false_string='false', **kwargs):
+        self.true_string = true_string
+        self.false_string = false_string
+        super(_BoolConfigValue, self).__init__(dict_key, **kwargs)
+
+    def __set__(self, instance, value):
+        if isinstance(value, compat.string_types):
+            if value.lower() == self.true_string:
+                value = True
+            elif value.lower() == self.false_string:
+                value = False
+        instance._values[self] = bool(value)
+
+
+class Config(object):
+    _NO_VALUE = object()  # sentinel object
+
+    app_name = _ConfigValue('APP_NAME', validators=[lambda val: re.match('.*', val)])
+    secret_token = _ConfigValue('SECRET_TOKEN')
+    servers = _ListConfigValue('SERVERS', default=['http://localhost:8200'])
+    include_paths = _ListConfigValue('INCLUDE_PATHS')
+    exclude_paths = _ListConfigValue('EXCLUDE_PATHS')
+    filter_exception_types = _ListConfigValue('FILTER_EXCEPTION_TYPES')
+    timeout = _ConfigValue('TIMEOUT', type=float)
+    hostname = _ConfigValue('HOSTNAME', default=socket.gethostname())
+    auto_log_stacks = _BoolConfigValue('AUTO_LOG_STACKS', default=True)
+    keyword_max_length = _ConfigValue('KEYWORD_MAX_LENGTH', type=int, default=1024)
+    transport_class = _ConfigValue('TRANSPORT_CLASS', default='elasticapm.transport.http_urllib3.AsyncUrllib3Transport')
+    processors = _ListConfigValue('PROCESSORS', default=[
+        'elasticapm.processors.sanitize_stacktrace_locals',
+        'elasticapm.processors.sanitize_http_request_cookies',
+        'elasticapm.processors.sanitize_http_headers',
+        'elasticapm.processors.sanitize_http_wsgi_env',
+        'elasticapm.processors.sanitize_http_request_querystring',
+        'elasticapm.processors.sanitize_http_request_body',
+    ])
+    traces_send_frequency = _ConfigValue('TRACES_SEND_FREQ', type=int, default=60)
+    async_mode = _BoolConfigValue('ASYNC_MODE', default=True)
+    instrument_django_middleware = _BoolConfigValue('INSTRUMENT_DJANGO_MIDDLEWARE', default=True)
+    transactions_ignore_patterns = _ListConfigValue('TRANSACTIONS_IGNORE_PATTERNS', default=[])
+    app_version = _ConfigValue('APP_VERSION')
+    disable_send = _BoolConfigValue('DISABLE_SEND', default=False)
+
+    def __init__(self, config_dict=None, env_dict=None, default_dict=None):
```

Comment:
Why don't you set `{}` and `os.environ` as default values instead of setting to `None` and then checking for `None`?

Reply:
That's Python Gotcha Number One ;p 

http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

Reply:
that's interesting, thank you for pointing it out (that's also why I mentioned that I reviewed as good as my python skills allowed ..)

Reply:
No worries :) It's not the number 1 gotcha because it would be stupid to make the mistake, it's number 1 gotcha because everybody falls into the trap sooner or later.